lib/rules/no-promise-without-done-fail.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+/**
+ * Determines whether a function call is a Promise.property() call
+ * @param {ASTNode} node A CallExpression node
+ * @returns {boolean} `true` if the call is a Promise.property() call
+ */
+function hasIdentifier (node, property) {
+  return node.callee &&
+    node.callee.type === 'MemberExpression' &&
+    node.callee.property.type === 'Identifier' &&
+    node.callee.property.name === property
+}
+
+function insideReturnStatement (node) {
+  return node.parent.type === 'ReturnStatement'
+}
+
+function hasDoneFailInReject (node, asyncParam) {
+  var reject = node.arguments[1]
+  return reject &&
+    reject.type === 'MemberExpression' &&
+    reject.object.type === 'Identifier' && reject.object.name === asyncParam &&
+    reject.property.type === 'Identifier' && reject.property.name === 'fail'
+}
+
+function isPromiseThenWithoutCatch (node, asyncParam) {
+  return hasIdentifier(node, 'then') &&
+    !hasIdentifier(node.parent.parent, 'then') &&
```

Comment:
Would `parent.parent` always exist for this node?

Reply:
Yes